### PR TITLE
feat(responsive-images): implement sharp.js, generate webp-images

### PR DESCRIPTION
**Pull request by Wandkleister**
This PR will enable us to generate webp-images in 3 different sizes.

- Images will get generated when you run the "start" npm script
- I integrated sharp.js for image processing
- The generated images will be put to ./_site/assets/images in 3 directories: retina, desktop and mobile

### CODE REVIEW

**Review by tam-kis — APPROVED** [review]

File: sharp.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
+const fs = require("fs");
+const glob = require("glob");
```
Comment (timbru31) [inline sharp.js:2]: Isn't this package missing in the package.json?!

File: sharp.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+			.then((metadata) => {
+				// Resize the image to a width specified by the `percent` value and output as PNG
+				return image
+					.resize(Math.round(metadata.width * (resize.percent / 100)))
+					.toFormat("webp")
+					.webp({ force: true })
+					.toFile(`${resize.dist}/${filename}`)
+					.catch((err) => {
+						console.log(err);
+					});
```
Comment (timbru31) [inline sharp.js:75]:
```suggestion
					.catch(console.error);
```

File: sharp.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+			})
+			.then(() => {
+				console.info("Processed image:", filename);
+			})
+			.catch((err) => {
+				console.log(err);
+			});
```
Comment (timbru31) [inline sharp.js:82]:
```suggestion
			.then(() => console.info("Processed image:", filename))
			.catch(console.error));
```

File: sharp.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+	// Find all files matching the glob patterns specified in `src`
+	let files = glob.sync(format.src);
+
+	files.forEach((file) => {
+		let filename = path.basename(file);
+		const image = sharp(file);
+		// Convert to WebP via Sharp's inferencing automatically of extensions
+		image.toFile(`${format.dist}/${filename.replace("png", format.format)}`).catch((err) => {
+			console.log(err);
+		});
```
Comment (timbru31) [inline sharp.js:106]:
```suggestion
		image.toFile(`${format.dist}/${filename.replace("png", format.format)}`).catch(console.error);
```

**Review by timbru31 — CHANGES_REQUESTED** [review]
Some minor style nits.

File: sharp.js (hunk 1 above)
Comment (Wandkleister) [reply sharp.js:2]: Yes it was missing! Added it ✅

File: package.json [hunk 5]
```diff
@@ -17,7 +17,8 @@
     "responsiveImages": "rimraf ./_site/assets/images/ && node sharp.js"
   },
   "dependencies": {
-    "@glidejs/glide": "^3.4.1"
+    "@glidejs/glide": "^3.4.1",
+    "glob": "latest"
```
Comment (timbru31) [inline package.json:21]: This should be a devDep and please use an actual version, i.e., `^7.1.6`

**Review by timbru31 — APPROVED** [review]

**Pull request merged**